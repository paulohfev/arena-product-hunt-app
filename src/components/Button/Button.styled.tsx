import styled from 'styled-components'

export const ButtonWrapper = styled.button<{ $backgroundColor: string; $border?: string; $color: string }>`
  background-color: ${({ $backgroundColor }) => $backgroundColor};
  border: ${({ $border }) => ($border ? $border : 'none')};
  border-radius: 8px;
  cursor: pointer;
  color: ${({ $color }) => $color};
  font-weight: 600;
  padding: 16px;
  transition: all 0.2s;
  width: 325px;

  &:disabled {
    cursor: not-allowed;
    opacity: 0.5;
  }

  &:hover {
    transform: translateY(-2px);
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  }
`
