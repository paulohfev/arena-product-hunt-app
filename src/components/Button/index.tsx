import { ButtonWrapper } from './Button.styled'

export type ButtonProps = {
  backgroundColor: string
  border?: string
  color: string
  children: React.ReactNode
  disabled?: boolean
  onClick?: () => void
}

const Button: React.FC<ButtonProps> = ({ backgroundColor, border, color, children, disabled, onClick }) => {
  return (
    <ButtonWrapper
      $backgroundColor={backgroundColor}
      $border={border}
      $color={color}
      disabled={disabled}
      onClick={onClick}
    >
      {children}
    </ButtonWrapper>
  )
}

export default Button
