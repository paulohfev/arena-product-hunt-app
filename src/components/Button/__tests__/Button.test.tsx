import { render } from '@testing-library/react'

import Button from '../index'

describe('Button', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <Button backgroundColor='white' color='black' border='1px solid #e0e0e0' disabled>
        Click me
      </Button>
    )
    expect(container).toMatchSnapshot()
  })
})
