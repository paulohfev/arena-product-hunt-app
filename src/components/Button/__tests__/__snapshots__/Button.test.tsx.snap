// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<div>
  <button
    class="sc-4e78a862-0 eKfrzH"
    disabled=""
  >
    Click me
  </button>
</div>
`;
